sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -103,4 +110,19 @@ case class UserDefinedFunction protected[sql] (
       udf
     }
   }
+
+  /**
+   * Updates UserDefinedFunction to non-deterministic.
+   *
+   * @since 2.3.0
+   */
+  def nonDeterministic(): UserDefinedFunction = {
```

Comment:
The default is `deterministic`. When changing it to non-deterministic, users can call `nonDeterministic()`. 

It might be rare to call `.withDeterministic(deterministic = true)`.